Definition retrieval endpoint

ecm68 opened the pull request:
`Main.py` and `controller.py` have been edited to retrieve a definition, given a term. 

Run the app from the repo root using `uvicorn main:app --reload`, then go to `http://127.0.0.1:8000/define/{term}`. (Fill in a term -- "humerus" for example). It will show the definition list in dictionary format. I opted to preserve the dictionary format for now, as opposed to converting to a list, so that we can continue to track the definition sources. **Let me know if we need list (not dict) format.**

The current behavior loads the term and definition JSONs each time the endpoint is called, as per the discussion Jon and I had on 5/20. This is something to look out for if it slows our app significantly once we have end-to-end functionality.

The expected outputs are:
A dictionary of format `{source1: def1, source2: def2, ...}`
- If the given term is defined in our dataset
An empty dictionary `{}`
- If the given term is in our jargon list but isn't defined
None
- If the given term is not in our jargon list

NOTE: This PR does not contain unit tests. I am hoping to start them after submitting this. :)

**Semantic Groupings**:
- This PR also adds a new file `semantic_groups.txt`, which is the list of semantic groups from UMLS that are included in our data (`term_to_cui.json`).

Commit:
added def retrieval endpoint

Commit:
added kept groupings file

jonjiang1 commented on main.py:
```diff
@@ -9,4 +9,10 @@ def read_root():
+
+@app.get("/define/{term}")
+def define(term: str):
+    term_file = "data/term_to_cui.json"
```
Since these are defined as static paths in retrieval.py do we need these here? Might be able to simplify this

jonjiang1 commented on model/term_def_retrieval/retrieval.py:
```diff
@@ -0,0 +1,51 @@
+
+
+
+def get_definition(sample: str, term_to_cui_file: str = "../../data/term_to_cui.json", 
```
Is it possible to change these paths to absolute ones instead of relative ones? That way if things were to move around we wouldn't have to worry about maintaining this.

Commit:
changed rel paths to abs

ecm68 commented:
@jonjiang1 - I added some code to retrieval.py which should find the data files' absolute paths, and I removed the data path arguments from all 3 files (`main.py`, `model/controller.py`, `retrieval.py`). 

I was having a hard time thinking of ways to find the data in the case that the model scripts get moved, so these changes aren't perfect:
- Assumes that the current working directory will be `med-jargon-explain-inator/` or lower
- Assumes that the paths to the data from the root is strict (always `.../med-jargon-explain-inator/data/filename.json`)

Let me know if either (or both) of these assumptions are counterintuitive, and I can try to fix that and push again. I just wasn't sure if these changes are on the right track!

jonjiang1 commented:
@ecm68 seems like sound reasoning to me! if there's more moves then we'll need to adjust but i doubt it at this point. LGTM!
